apps/x509.c

```diff
@@ -842,9 +901,15 @@ int x509_main(int argc, char **argv)
                     if (Upkey == NULL)
                         goto end;
                 }
-
+#ifndef OPENSSL_NO_SM2
+                if (EVP_PKEY_id(Upkey) == EVP_PKEY_SM2) {
```

Comment:
This seems not necessary anymore in 3.0.0. I recall @levitte resolved this to automatically set the SM2 tag when loading a key.

Reply:
Yup

Reply:
fixed in last commit:-)

Reply:
A side note: the PR @levitte created during the Chinese lunar new year is https://github.com/openssl/openssl/pull/10942 ;-)